veg2hab/vegkartering.py

```diff
@@ -944,6 +991,110 @@ def apply_wwl(
             wwl.toevoegen_VvN_aan_List_VegTypeInfo
         )
 
+    @staticmethod
+    def _vegtypeinfo_to_multi_col(vegtypeinfos: List[VegTypeInfo]) -> pd.Series:
+        result = pd.Series()
+        for idx, info in enumerate(vegtypeinfos, 1):
+            result[f"SBB{idx}"] = ",".join(
+                str(sbb) for sbb in info.SBB
+            )  # convert to pandas string..
+            result[f"VvN{idx}"] = ",".join(str(vvn) for vvn in info.VvN)
+            result[f"perc{idx}"] = info.percentage
+        return result
+
+    def to_editable_vegtypes(self) -> gpd.GeoDataFrame:
+        # unpack the vegtypeinfo
+        vegtypes_df = self.gdf["VegTypeInfo"].apply(self._vegtypeinfo_to_multi_col)
+        str_columns = {
+            name: "string"
+            for name in vegtypes_df.columns
+            if name.startswith("SBB") or name.startswith("VvN")
+        }
+        perc_columns = {
+            name: float for name in vegtypes_df.columns if name.startswith("perc")
+        }
+        vegtypes_df = vegtypes_df.astype({**str_columns, **perc_columns})
+        vegtypes_df[list(str_columns.keys())] = vegtypes_df[
+            list(str_columns.keys())
+        ].fillna("")
+
+        # move and rename vegtype info column to the end
+        gdf = self.gdf.rename(columns={"VegTypeInfo": "_VegTypeInfo"})
+        gdf = pd.concat([gdf, vegtypes_df], axis=1)
+
+        gdf["_VegTypeInfo"] = (
+            gdf["_VegTypeInfo"].apply(VegTypeInfo.serialize_list).astype("string")
+        )
+
+        column_order = [
+            *self.PREFIX_COLS,
+            *vegtypes_df.columns,
+            "_VegTypeInfo",
+            *self.POSTFIX_COLS,
+        ]
+
+        gdf = gdf[column_order]
+
+        # for some dumb reason ARCGis handles columns that begin with a _ or a number
+        # really badly.
+        rename_cols = {
+            col: "INTERN" + col for col in column_order if col.startswith("_")
```

Comment:
Waarom noem je de kolom eerst _VegTypeInfo? Dat is de enige _ kolom hier nog toch?